core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1139,9 +1139,13 @@ object Log {
       throw exception(dir)
 
     val name: String =
-      if (dirName.endsWith(DeleteDirSuffix)) dirName.substring(0, dirName.indexOf('.'))
+      if (dirName.endsWith(DeleteDirSuffix)) dirName.substring(0, dirName.lastIndexOf('.'))
       else dirName
 
+    if (!name.contains('-')) {
```

Comment:
Good point. That commit which brought in that regex check, in trunk, was self contained. So I have cherry-picked it and included it in this PR.